assets/src/edit-story/components/form/numeric.js

```diff
@@ -140,6 +180,15 @@ function Numeric({
 
   const { focused, handleFocus, handleBlur } = useFocusAndSelect(ref);
 
+  useEffect(() => {
+    if (targetFormRef.current) {
+      submitForm(targetFormRef.current);
+      // Resetting to null after dispatch, it'll get
+      // re-assigned on component blur
+      targetFormRef.current = null;
+    }
+  }, [value]);
```

Comment:
In order to process a potential final update from `onBlur`,  have this useEffect that changes every time value changes from a call to `onChange`.  The targetFormRef only gets set within the blur method, so if it's set, that means we want to dispatch and then we immediately reset it so further updates won't trigger anything.